aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/emitters/UDPEmitter.java

```diff
@@ -58,17 +58,17 @@ public boolean sendSubsegment(Subsegment subsegment) {
         if (logger.isDebugEnabled()) {
             logger.debug(subsegment.prettyStreamSerialize());
         }
-        return sendData((PROTOCOL_HEADER + PROTOCOL_DELIMITER + subsegment.streamSerialize()).getBytes());
+        return sendData((PROTOCOL_HEADER + PROTOCOL_DELIMITER + subsegment.streamSerialize()).getBytes(), subsegment);
     }
 
-    private boolean sendData(byte[] data) {
+    private boolean sendData(byte[] data, Entity entity) {
         DatagramPacket packet = new DatagramPacket(sendBuffer, DAEMON_BUF_RECEIVE_SIZE, config.address);
         packet.setData(data);
         try {
             logger.debug("Sending UDP packet.");
             daemonSocket.send(packet);
         } catch (IOException e) {
-            logger.error("Exception while sending segment over UDP.", e);
+            logger.error("Exception while sending entity over UDP: " + entity.prettySerialize(), e);
```

Comment:
As I said in the first comment, could you change this one to only print the segments/subsegments?

Reply:
Only printing the current segment/sub-segment id won't help us much I think, as the problem is in our use case typically in an underlying sub-segment.